src/ngraph/op/broadcastdistributed.cpp

```diff
@@ -0,0 +1,45 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/broadcastdistributed.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+op::BroadcastDistributed::BroadcastDistributed(const shared_ptr<Node>& arg)
+    : Op("BroadcastDistributed", check_single_output_args({arg}))
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::BroadcastDistributed::validate_and_infer_types()
+{
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(0).is_dynamic() ||
+                              get_input_element_type(0) == element::f32 ||
+                              get_input_element_type(0) == element::f64,
```

Comment:
It's funny we have this.  Who's training with double precision? 😋

Reply:
iT's thErE in tHe fraMewoRKS